pkg/model/components/kubelet.go

```diff
@@ -61,6 +61,11 @@ func (b *KubeletOptionsBuilder) BuildOptions(o interface{}) error {
 	clusterSpec.Kubelet.ClusterDomain = clusterSpec.ClusterDNSDomain
 	clusterSpec.Kubelet.NonMasqueradeCIDR = clusterSpec.NonMasqueradeCIDR
 
+	// Set secure kubelet on by default when the user has not set this flag
+	if clusterSpec.Kubelet.AnonymousAuth == nil {
+		clusterSpec.Kubelet.AnonymousAuth = fi.Bool(false)
```

Comment:
Yes .. but this would force all clusters to suddenly use mutal tls on the kubelet api, so it would need to be highlighted vigorously not to surprise. I know @justinsb did some work on state to detect is new cluster or updating a cluster .. i'm not sure if this information is communicated here though ... @justinsb?

Reply:
@gambol99 to see if I understand what you are saying, and thanks for the review.  We would need to determine how an update would work with this.  I think this is an ok change for a new cluster, but with an existing cluster, this would cause downtime.  When the masters change to using TLS first during an update, not sure how the nodes would respond.  @audip have you tested an upgrade?

@gambol99 we need rolling update e2e 🙊

Reply:
I have not tested the upgrade yet.